Add board waveshare-rp2040-lcd-0.96 (#513)
rp-rs/rp-hal: boards/waveshare-rp2040-lcd-0_96/README.md

a new `Pins` structure. This will set up all the GPIOs for any on-board
devices. See the [examples](./examples) folder for more details.

## Examples

### General Instructions

To compile an example, clone the _rp-hal_ repository and run:

```console
rp-hal/boards/waveshare-rp2040-lcd-0_96 $ cargo build --release --example <name>
```

You will get an ELF file called
`./target/thumbv6m-none-eabi/release/examples/<name>`, where the `target`
folder is located at the top of the _rp-hal_ repository checkout. Normally
you would also need to specify `--target=thumbv6m-none-eabi` but when
building examples from this git repository, that is set as the default.

If you want to convert the ELF file to a UF2 and automatically copy it to the
USB drive exported by the RP2040 bootloader, simply boot your board into
bootloader mode and run:

```console
rp-hal/boards/waveshare-rp2040-lcd-0_96 $ cargo run --release --example <name>
```

If you get an error about not being able to find `elf2uf2-rs`, try:

```console
$ cargo install elf2uf2-rs, then repeating the `cargo run` command above.
```

### [waveshare_rp2040_lcd_demo](./examples/waveshare_rp2040_lcd_demo.rs)

> Draws a red and green line with a blue regtangle.
> After that is fills the screen line for line, that end it starts over with an
> other colour, RED, GREEN and BLUE.

Reviewer: ```suggestion
Draws a red and green line with a blue rectangle.
After that it fills the screen line for line, at the end it starts over with
another colour, RED, GREEN and BLUE.
```
